packages/firestore/src/local/persistence.ts

```diff
@@ -285,3 +290,42 @@ export interface Persistence {
     ) => PersistencePromise<T>
   ): Promise<T>;
 }
+
+/**
+ * Interface implemented by the LRU scheduler to start(), stop() and restart
+ * garbage collection.
+ */
+export interface GarbageCollectionScheduler {
+  readonly started: boolean;
+  start(localStore: LocalStore): void;
+  stop(): void;
+}
+
+/**
+ * Provides all persistence components for Firestore. Consumers have to invoke
+ * configure() once before accessing any of the individual components.
+ */
+export interface PersistenceProvider {
+  /**
+   * True if the underlying implementation supports durable storage via
+   * `enablePersistence()`.
+   */
+  readonly isDurable: boolean;
```

Comment:
Couldn't we have `configure` throw if the `settings` include enabling persistence but the provider doesn't actually support it?

Reply:
That's how I ended up implementing the new error semantics for the MemoryPersistenceProvider.